securedrop_client/gui/main.py

```diff
@@ -117,11 +132,21 @@ def logout(self):
         """
         self.tool_bar.set_logged_out()
 
+    def on_source_changed(self):
+        """
+        React to when the selected source has changed.
+        """
+        source_item = self.main_view.source_list.currentItem()
+        source_widget = self.main_view.source_list.itemWidget(source_item)
+        self.show_conversation_for(source_widget.source)
+
     def show_conversation_for(self, source=None):
         """
         TODO: Finish this...
         """
         conversation = ConversationView(self)
+        conversation.add_message('Source name: {}'.format(
+                                 source.journalist_designation))
```

Comment:
since source can be `None`, we should put a `if source: ` above this line

Reply:
Actually, there *should* be a source (no need for the default `None`).